app/src/main/java/org/oppia/app/player/audio/AudioViewModel.kt

```diff
@@ -44,16 +60,41 @@ class AudioViewModel @Inject constructor(
     processPlayStatusLiveData()
   }
 
-  fun setVoiceoverMappings(map : Map<String, Voiceover>) {
-    voiceoverMap = map
+  val showSeekBar = ObservableField(true)
+
+  fun getExplorationById(exploreId: String, stateId: String) {
+    explorationId = exploreId
+    val explorationResultLiveData = explorationDataController.getExplorationById(explorationId)
+    processExplorationLiveData(explorationResultLiveData).observeForever {
+      exploration = it
+      setVoiceoverMappingsByState(stateId)
+    }
   }
 
-  fun setExplorationId(id: String) {
-    explorationId = id
+  fun setVoiceoverMappingsByState(stateId: String) {
+    val state = exploration.statesMap[stateId] ?: State.getDefaultInstance()
+    val contentId = state.content.contentId
+    voiceoverMap = (state.recordedVoiceoversMap[contentId] ?: VoiceoverMapping.getDefaultInstance()).voiceoverMappingMap
+    languages = voiceoverMap.keys.toList().map { it.toLowerCase(Locale.getDefault()) }
+    isLanguagesEmpty.set(languages.isEmpty())
+    if (languages.any { it == defaultLanguage }) {
+      setAudioLanguageCode(defaultLanguage)
+    } else if (languages.any { it == selectedLanguageCode }) {
+      setAudioLanguageCode(selectedLanguageCode)
+    } else {
+      if (selectedLanguageCode.isEmpty() && languages.isNotEmpty()) {
+        setAudioLanguageCode(languages.first())
```

Comment:
Per the email thread, shouldn't we select English by default if it's available, and then the first language otherwise?